tools/cmake/app.cmake

```diff
@@ -19,3 +19,16 @@ endif()
 include(${MBED_PATH}/tools/cmake/profile.cmake)
 
 enable_language(C CXX ASM)
+
+# Find Python and needed packages
+find_package(Python3)
+include(${CMAKE_CURRENT_LIST_DIR}/CheckPythonPackage.cmake)
+check_python_package(intelhex HAVE_INTELHEX)
```

Comment:
we should read requirement.txt in tools/cmake as we might need few packages more (we are adding post build hooks that rely on some security modules).

Reply:
The  below could check if any of req. are not installed, print a warning otherwise we got all we need

Reply:
The `requirements.txt` in the root directory should also be checked as it is the one that has the required packages for post build operations..

Reply:
This is possible but it's a bit harder because it means CMake needs to parse the "platform_system==XXXX" declarations that are used in that file.

Reply:
> The requirements.txt in the root directory should also be checked as it is the one that has the required packages for post build operations..

Let's keep Cmake reqs separate and do proper clean up of deps (drop in replacement once the old tools are removed).